sdlib/d/gc/memmap.d

```diff
@@ -88,12 +88,16 @@ void pages_zero(void* addr, size_t size) {
 
 void pages_hugify(void* addr, size_t size) {
 	auto ret = madvise(addr, size, Madv.HugePage);
-	assert(ret == 0, "madvise failed!");
+	// MADV_HUGEPAGE is not always supported by the kernel, so ignore
+	// failed return values.
+	// assert(ret == 0, "madvise failed!");
```

Comment:
I think a more appropriate way to handle this would be to check whether rte is 0 or `EINVAL`, as this is the error we get. If we assert that addr and size are properly aligned, we should catch most program error anyways.

Reply:
Ok it is in errno, so it is anoying.

Please change the explanation then, mention that calling madvise with MADV_HUGEPAGE when it is isn't support result in a EINVAL error. At least whoever deals with that later will know what we know (and there are good chances that that person in one of us anyways).